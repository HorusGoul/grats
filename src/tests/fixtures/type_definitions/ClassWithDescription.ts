/**
 * The root of all evil.
 *
 * @gqlType
 */
export default class SomeType {
  /** @gqlField */
  hello: string;
}
