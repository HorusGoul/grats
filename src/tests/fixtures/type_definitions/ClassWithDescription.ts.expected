-----------------
INPUT
----------------- 
/**
 * The root of all evil.
 *
 * @gqlType
 */
export default class SomeType {
  /** @gqlField */
  hello: string;
}

-----------------
OUTPUT
-----------------
-- SDL --
directive @asyncIterable on FIELD_DEFINITION

directive @propertyName(name: String!) on FIELD_DEFINITION

directive @exported(tsModulePath: String!, functionName: String!, argCount: Int!) on FIELD_DEFINITION

"""The root of all evil."""
type SomeType {
  hello: String
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
function getSchema(): GraphQLSchema {
    const SomeTypeType: GraphQLObjectType = new GraphQLObjectType({
        name: "SomeType",
        description: "The root of all evil.",
        fields() {
            return {
                hello: {
                    name: "hello",
                    type: GraphQLString
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [SomeTypeType]
    });
}
export { getSchema };
